Chrysalis-Papilio/Soen490-Papilio-Website
# Api layer

## General Information

**Issue number**: Papilio-64

**Task description:** 
1. Add an API layer to abstract the fetch method within the React application.
- Create abstract method
- Replace fetch method within the components with the abstraction
- Test the abstraction and make the appropriate correction in component tests

2. Add employee context to save the information.

## Manual Testing
To test the profile creation you have to create your own business then visit the different page of the dashboard 

1. Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)
2. start the backend
```npm run build && npm start```
3. start the website
```npm start```
4. on the main page add the businessId then click on create business
5. enter the profile information then click next
6. enter the admin root account information then click next
7. click submit
8. Visit the different page of the dashboard

## Testing
To run all the unit tests, run the following code: ```npm test```
To make sure that the tests have reach the appropriate level of coverage, run the following code: ```npm run coverage```
